backend/entityservice/database/selections.py

```diff
@@ -260,32 +260,36 @@ def get_dataprovider_id(db, update_token):
     return query_db(db, sql_query, [update_token], one=True)['id']
 
 
-def get_bloomingdata_column(db, dp_id, column):
-    assert column in {'ts', 'token', 'file', 'state', 'count'}
+def get_bloomingdata_columns(db, dp_id, columns):
+    for column in columns:
+        assert column in {'ts', 'token', 'file', 'state', 'count', 'encoding_size'}
     sql_query = """
         SELECT {} 
         FROM bloomingdata
         WHERE dp = %s
-        """.format(column)
+        """.format(', '.join(columns))
     result = query_db(db, sql_query, [dp_id], one=True)
     if result is None:
         raise DataProviderDeleted(dp_id)
     else:
-        return result[column]
+        if len(columns) > 1:
+            return [result[column] for column in columns]
+        else:
+            return result[columns[0]]
```

Comment:
Not 100% enjoying returning a different type for convenience when there is only one column. I get it & I've done it I just don't like it.